README.md
#Atomic Developer Bundle (ADB)

##Purpose and Goal

To provide a ready to use development environment for developers creating container-based applications.  This environment will support a developer in packaging their application service components into containers and defining the entire application for ease of installation and testing.  While the environment can be used from the beginning of the development process, it is expected that most users will have their core code working prior to starting the containerization process.  You may find reading the original project [motivation](docs/motivation.md) useful.

Ultimately, the project expects to deliver a virtual machine that supports:

* **Providing docker support to unsupported platforms (i.e. Microsoft Windows, Mac OS X, etc.)**
* Enabling host-based development tools such as Eclipse to access the build environments inside the ADB
* **Kubernetes orchestration for local testing of applications**
* Platform as a Service (PAAS) local testing and remote integration for [OpenShift](https://github.com/openshift).
* Enabling a workflow for define once, run anywhere that allows configuration for Openshift to be resused for Kubernetes
* **Application definition using the [Nulecule](https://github.com/projectatomic/nulecule) specification**
* The [Nulecule DevAssistant](https://github.com/devassistant/dap-nulecule) to make defining applications easier by providing a scaffold
* Application definition enablement using the [Atomic App](https://github.com/projectatomic/atomicapp) Nulecule reference implementation
* [atomicapp-builder](https://github.com/bkabrda/atomicapp-builder) to drive atomicapp builds within the ADB
* This ability to push applications into the [CentOS Community Container Pipeline](http://wiki.centos.org/ContainerPipeline) via a pull request to the [CentOS Community Container Pipeline index](https://github.com/kbsingh/cccp-index)
* Supporting multi-node local testing and building using solutions like [Oh-my-vagrant](https://github.com/purpleidea/oh-my-vagrant)

**Note: The most recent release is Beta 2.  Bolded items above have been started or are working in this beta.**

Boxes are built using the CentOS powered [Community Build System](http://cbs.centos.org/koji/).  Boxes are delivered via Hashicorp's [Atlas](https://atlas.hashicorp.com/atomicapp/boxes/dev).  While Project Atomic already provides vagrant boxes for CentOS and Fedora, we can not reuse these as we have specific build requirements not enabled by default.

##Documentation

* [Building the Vagrant box](docs/build.rst)
* [Quickstart to the Atomic App vagrant box](docs/quickstart.rst)
* [Running atomicapp](docs/runningatomicapp.rst)
* [Update the box](docs/updating-thebox.rst)
* [Whats inside the Vagrant box](docs/whatsinside.rst)

##How to contribute?

* You are welcome to raise issues, send pull requests.

###Contribute to documentation

* Documentations is in [reStructuredText](http://docutils.sourceforge.net/rst.html) as it integrates well with [readthedocs](https://readthedocs.org) project.
* Here is an [online reStructuredText editor](http://rst.ninjs.org) for reference.

##Communication channels

* IRC: #nulecule (On Freenode)
* Mailing List: [container-tools@redhat.com](https://www.redhat.com/mailman/listinfo/container-tools)


CHANGELOG.md
# Changelog

## v2.3.1 Sep 08, 2016
- Fix #534. Fixing the CDK OpenShift kickstart file to cache ose-pod container image @LalatenduMohanty
- Increasing the /etc/os-release version to 2.3.0 @LalatenduMohanty
- Fix #532 disable rsync and use sshfs by default in CDK Kubernetes Vagrantfile @praveenkumar
- Fix #528 Failed to start Etcd Server K8s Vagrantfile @praveenkumar
- Revert landrush plugin for CDK Vagrantfiles @praveenkumar
- Fix #444, added CDK product page link and cleaned up formatting issues @Preeticp
- Fix #509 Proxy support k8s vagrantfile @praveenkumar
- Fix #517 rsync: connection unexpectedly closed @praveenkumar
- Updated centos-k8s Vagrantfile to start through service-manager hook @budhrg
- Adding Budh Ram Gurung(@budhrg) as maintainer @budhrg
- Adding suspend_mode = 'managedsave' in libvirt configuration @rimolive

## v2.3.0 Aug 10, 2016
- Add Landrush as plugin requirement in Vagrantfile @praveenkumar
- Unify Vagrantfile for different Providers @praveenkumar
- Fixes issue #488. Using ENV variables for proxy @LalatenduMohanty
- Issue #461 Enabling Landrush per default @hferentschik
- Fix a couple broken links @shawnlower
- Updated latest openshift tag @praveenkumar
- Fix #478 Vagrantfile update for proxy support @praveenkumar
- Fix #379 added note about using the latest origin image and corrected installation doc link @preeticp
- Fix #460 Doc update for proxy support @praveenkumar
- Fix #472, #473 and #477 @preeticp
- marathon: embed ansible playbook in Vagrantfile @dustymabe
- marathon: setup so that ports from host can be shared @dustymabe

## v2.2.2 Jul 18, 2016
- Update README to add Mesos-Marathon and cache notes @bexelbie
- Bumping /etc/os-release for ADB release 2.2.2 @@LalatenduMohanty
- Fix #463 Update ose tag to 3.2.1.4 in CDK @praveenkumar
- Install atomicapp as a part of provisioning in Mesos Marathon Vagrantfile @dharmit

## v2.2.1 Jun 23, 2016
- Fixes Issue #325; Improved Readme installation instructions and general clean up @preeticp
- Bumping up os-release file for ADB 2.2.1 release @LalatenduMohanty
- Improved Installation workflow and general clean up. @preeticp
- Adding "-provider hyperv" to "vagrant up" in README files for Hyper-V @LalatenduMohanty
- Adding Vagrantfiles for HyperV. @LalatenduMohanty
- Adding HyperV support to the CDK kickstart file. @LalatenduMohanty
- Updating README_CDK_ZIP with vagrant-sshfs information @LalatenduMohanty
- Adding some blogs and youtube videos from web to readme @LalatenduMohanty
- Fix #428 Success message after vagrant up for k8s-singlenode-setup  @praveenkumar
- Issue #272 Disabling /vagrant folder sync via rsync @hferentschik
- Issue #336 Mounting users home directory into VM using vagrant-sshfs @hferentschik
- Update OpenShift Origin default version to 1.2.0  @praveenkumar

## v2.2.0 June 9, 2016
- Update variant information @praveenkumar
- Can not change securityContext in k8s @praveenkumar
- Fix #422 Failed to get pwuid struct: user: unknown userid @praveenkumar
- Reduce vagrant box size #176 @praveenkumar
- rhel-k8s Vagrantfile: fix in getting credentials @optak
- Changes for Release 2.2.0 @LalatenduMohanty
- Add openshift tag information to kickstart instead in sccli/openshift_options @praveenkumar
- Fix #415 add yum clean all step to kickstart file to remove yum local cache @praveenkumar
- Fix #414 kube-apiservice shouldn't be part of kickstart file @praveenkumar
- Remove Public_Host variable from OpenShift Vagrantfile @praveenkumar
- Fix #407 for "Error: dev/null: No such file or directory" @praveenkumar
- Fixes #347 Run provisioners for every vagrant up. @LalatenduMohanty
- Updated cockpit guide to start cockpit @budhrg
- Fixes #251: Adding sccli/ADB Utils repo link in the ADB Usage doc. @preeticp
- Docker SELinux context not set properly when building vagrant box.  @praveenkumar
- Add https://titanpad.com/adbmeeting to README @bexelbie

## v2.1.0 May 19, 2016
- Fix #353: Add VSM config to docker setup and removed Vagrantbox Readme @praveenkumar
- Bumping /etc/os-release to 2.1.0 @LalatenduMohanty
- Remove development group from kickstart @praveenkumar
- Updated cockpit guide to start cockpit @budhrg

## v2.0.0 Apr 28, 2016
- Fix #351 Suppress logs from systemctl enable for kubernetes @praveenkumar
- Remove the workaround for downgrading python-docker-py @LalatenduMohanty
- Adding openshift2nulecule to ADB @LalatenduMohanty
- Update os-release to 2.0.0 @LalatenduMohanty
- Fix #103: build_tools/kickstarts/rhel-7-cdk-vagrant.ks @praveenkumar
- Fix #345: Suppress logs of openssl genrsa on Vagrant up for Kubernetes @budhrg
- Fix #342: Use systemctl to start openshift service in CDK OSE Vagrantfile @LalatenduMohanty
- Fix #334: Disables openshift service for CDK k8s Vagrantfile @navidshaikh
- Fix-256: Add check for vagrant-registration plugin @budhrg
- Refactored code for checking required plugins for CentOS files @budhrg
- Update README for RHEL component @praveenkumar
- Fix #326 (dev part) disk partition fix @praveenkumar
- Fix #332 Vagrantfile var name fix @optak
- Removed kube-apiserver service file duplication @praveenkumar
- Added Notes on Default for ADB/CDK; Docker version mismatch; and moved relevant VSM steps.@preeticp
- Fix #228 Add Cockpit @bexelbie
- Fix #251; Adding sccli/ADB Utils repo link in the ADB Usage doc @preeticp

## v1.8.0 Apr 5, 2016
- Update Vagrantfiles with RHEL files @LalatenduMohanty
- Update os-release to 1.8.0 @LalatenduMohanty
- Remove docker-registry from installed packages @praveenkumar
- Use copy rather than shell in ansible playbook for Mesos Vagrantfile @mscherer
- Adding fuse-sshfs to the ADB box @LalatenduMohanty

## v1.7.2 Mar 24, 2016
- Changes for release 1.7.2 @LalatenduMohanty
- OpenShift Vagrantfile mod to reflect new command @bexelbie

## v1.7.1 Mar 9, 2016
- Revert "Adding openshift2nulecule to ADB" @LalatenduMohanty
- Adding openshift2nulecule to ADB @LalatenduMohanty
- Update docs for Fedora libvirt @bexelbie
- Fixing minor typos @Preeticp

tests/standalone-rhel/test-base-rhel.sh
#!/bin/bash

export PATH="/bin:/usr/bin"
STATUS=0

echo "get the plugin installed"
vagrant plugin install vagrant-registration

echo "do we have any plugins now?"
exec 5>&1 && OUTPUT=$(vagrant plugin list | tee >(cat - >&5))
if [ $? -ne 0 ]; then
    STATUS=$?
    echo "No plugins at all:  Vagrant plugin test FAILED"
    exit $STATUS
fi

echo "do we have the vagrant-registration plugin?"
TEST=$(echo "$OUTPUT" | grep vagrant-registration)
STATUS=$?
if [ -z "$TEST" ]; then
    echo "Plugin vagrant-registration did not get installed:  Vagrant plugin test FAILED"
    echo "Output of vagrant plugin list:\n$OUTPUT\n"
    echo "Result of \"echo \$OUTPUT | grep vagrant-registration\":\n$TEST\n"
    exit $STATUS
fi

echo "let's pull over the vagrantfile to test"
rm ./Vagrantfile
ln -s ../../components/standalone-rhel/Vagrantfile ./

echo "let's try and bring the machine up"
exec 5>&1 && OUTPUT=$(vagrant up | tee >(cat - >&5))
STATUS=$?
if [ $? -ne 0 ]; then
    echo "Failed to bring up the machine: Launch FAILED"
    exit $STATUS
fi

echo "let's test that we can connect"
OUTPUT=$(vagrant ssh -c 'echo \"connected!\"' | tee >(cat - >&5))
STATUS=$?
TEST=$(echo "$OUTPUT" | grep connected)
if [ -z "$TEST" ]; then
    echo "Failed to connect to the machine: connect FAILED"
    exit $STATUS
fi

echo "let's test that we are subscribed"
OUTPUT=$(vagrant ssh -c 'sudo subscription-manager status' | tee >(cat - >&5))
STATUS=$?
TEST=$(echo "$OUTPUT" | grep Current)
if [ -z "$TEST" ]; then
    echo "We are not subscribed: subscribe FAILED"
    exit $STATUS
fi

echo "time to clean up"
vagrant destroy -f
rm -rf .vagrant.d Vagrantfile

echo "tests complete!"


tests/rhel-with-docker/test-rhel-docker.sh
#!/bin/bash

export PATH="/bin:/usr/bin"
STATUS=0

#get the plugin installed
vagrant plugin install vagrant-registration

#do we have any now?
exec 5>&1 && OUTPUT=$(vagrant plugin list | tee >(cat - >&5))
if [ $? -ne 0 ]; then
    STATUS=$?
    echo "No plugins at all:  Vagrant plugin test FAILED"
    exit $STATUS
fi

#do we have the reg plugin?
TEST=$(echo "$OUTPUT" | grep vagrant-registration)
STATUS=$?
if [ -z "$TEST" ]; then
    echo "Plugin vagrant-registration did not get installed:  Vagrant plugin test FAILED"
    echo "Output of vagrant plugin list:\n$OUTPUT\n"
    echo "Result of \"echo \$OUTPUT | grep vagrant-registration\":\n$TEST\n"
    exit $STATUS
fi

#lets pull over the files to test
cp -R ./../components/rhel-with-docker ./
cd rhel-with-docker/dev/

echo "incomplete test"
exit

#lets try and bring the machine up
OUTPUT=$(vagrant up | tee >(cat - >&5))
STATUS=$?
if [ $? -ne 0 ]; then
    echo "Failed to bring up the machine: Launch FAILED"
    exit $STATUS
fi

#lets test that we can connect
OUTPUT=$(vagrant ssh -c 'echo \"connected!\"' | tee >(cat - >&5))
STATUS=$?
TEST=$(echo "$OUTPUT" | grep connected)
if [ -z "$TEST" ]; then
    echo "Failed to connect to the machine: connect FAILED"
    exit $STATUS
fi

#lets test that we are subscribed
OUTPUT=$(vagrant ssh -c 'sudo subscription-manager status' | tee >(cat - >&5))
STATUS=$?
TEST=$(echo "$OUTPUT" | grep Current)
if [ -z "$TEST" ]; then
    echo "We are not subscribed: subscribe FAILED"
    exit $STATUS
fi

#time to clean up
vagrant destroy -f
rm -rf .vagrant.d Vagrantfile

echo "Test(s) complete!"


build_tools/build_scripts/do_vagrant_cbs_test.sh
#!/bin/sh
#To see all options in koji -p command check "koji -p cbs image-build --help"

koji -p cbs image-build \
  centos-7-adb 1.5 atomic7-adb-common-el7 \
  http://mirror.centos.org/centos/7/os/x86_64/ x86_64 \
  --release=1 \
  --distro RHEL-7.0 \
  --ksver RHEL7 \
  --kickstart=./build_tools/kickstarts/centos-7-adb-vagrant.ks \
  --format=qcow2 \
  --format=vsphere-ova \
  --format=rhevm-ova \
  --ova-option vsphere_ova_format=vagrant-virtualbox \
  --ova-option rhevm_ova_format=vagrant-libvirt \
  --ova-option vagrant_sync_directory=/vagrant \
  --repo http://mirror.centos.org/centos/7/extras/x86_64/\
  --repo http://mirror.centos.org/centos/7/updates/x86_64/\
  --repo http://cbs.centos.org/repos/atomic7-adb-common-testing/x86_64/os/\
  --scratch \
  --nowait \
  --disk-size=40



src/spellcheck.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

# Copyright © 2015  Praveen Kumar <kumarpraveen.nitdgp@gmail.com>
#
# This copyrighted material is made available to anyone wishing to use,
# modify, copy, or redistribute it subject to the terms and conditions
# of the GNU General Public License v.2, or (at your option) any later
# version.  This program is distributed in the hope that it will be
# useful, but WITHOUT ANY WARRANTY expressed or implied, including the
# implied warranties of MERCHANTABILITY or FITNESS FOR A PARTICULAR
# PURPOSE.  See the GNU General Public License for more details.  You
# should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation,
# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
#


# Usage python spellcheck.py <folder/file path>

"""
Sample Usage:
$ python spellcheck.py ../docs/
+128 ../docs/installing.rst : scl
Output format:
    +<line_number> <file_name> <misspelled_word>
If you use vim then directly copy/paste `vim +<line_number> <file_name>` and you
correct misspelled word fast.
You can add word to validwords file so it will added to custom dictionary which
passed to SpellChecker
"""

import os
from argparse import ArgumentParser
from enchant import DictWithPWL
from enchant.checker import SpellChecker

my_dict = DictWithPWL("en_US", "adb.dict")
checker = SpellChecker(my_dict)
file_extension = ['.md', '.txt', '.rst']

def list_files(rootdir):
    file_list = []
    for root, subdirs, files in os.walk(rootdir):
        for name in files:
            if os.path.splitext(name)[-1] in file_extension:
                file_list.append(os.path.join(root, name))
    return file_list

def check_spell(file_name):
    with open(file_name) as fh:
        for num, line in enumerate(fh, 1):
            checker.set_text(line)
            for err in checker:
                print "+%d %s : %s" % (num, file_name, err.word)

if __name__ == '__main__':
    parser = ArgumentParser(description='Spell checks using en_US dictionary')
    parser.add_argument('path', type=str,
                        help='Provide absoulte path of file/folder to spellcheck')
    args = parser.parse_args()
    file_list = list_files(args.path)
    for file_path in file_list:
        check_spell(file_path)


build_tools/kickstarts/cert-gen.sh
#!/bin/bash
# Generate certs pair for configuring TLS enabled docker daemon
. /etc/sysconfig/docker

if [ -e $DOCKER_CERT_PATH/ca.pem ]; then
  # Certificates already generated
  exit
fi

function randomString {
        # if a param was passed, it's the length of the string we want
        if [[ -n $1 ]] && [[ "$1" -lt 20 ]]; then
                local myStrLength=$1;
        else
                # otherwise set to default
                local myStrLength=8;
        fi

        dd if=/dev/urandom bs=1 2>/dev/null | tr -dc '[:alnum:]' | dd bs=1 count=$myStrLength 2>/dev/null
}

# Get a temporary workspace
dir=`mktemp -d`
cd $dir

# Get a random password for the CA and save it
passfile=tmp.pass
password=$(randomString 10)
echo $password > $passfile

# Generate the CA
openssl genrsa -aes256 -passout file:$passfile -out ca-key.pem 2048
openssl req -new -x509 -passin file:$passfile -days 365 -key ca-key.pem -sha256 -out ca.pem -subj "/C=/ST=/L=/O=/OU=/CN=example.com"

# Generate Server Key and Sign it
openssl genrsa -out server-key.pem 2048
openssl req -subj "/CN=example.com" -new -key server-key.pem -out server.csr
# Allow from routable local IP
extip=`ip addr show eth1 2>/dev/null | awk 'NR==3 {print $2}' | cut -f1 -d\/`
if [[ -z "$extip" ]]; then
      extip=`ip route get 8.8.8.8 | awk 'NR==1 {print $NF}'`
fi
extipfile=extfile.cnf
echo subjectAltName = IP:$extip > $extipfile
openssl x509 -req -days 365 -in server.csr -CA ca.pem -CAkey ca-key.pem -CAcreateserial -out server-cert.pem -passin file:$passfile -extfile $extipfile

# Generate the Client Key and Sign it
openssl genrsa -out key.pem 2048
openssl req -subj '/CN=client' -new -key key.pem -out client.csr
extfile=tmp.ext
echo extendedKeyUsage = clientAuth > $extfile
openssl x509 -req -days 365 -in client.csr -CA ca.pem -CAkey ca-key.pem -CAcreateserial -out cert.pem -extfile $extfile -passin file:$passfile

# Clean up

# set the cert path as configured in /etc/sysconfig/docker

## Move files into place
mv ca.pem $DOCKER_CERT_PATH
mv server-cert.pem $DOCKER_CERT_PATH
mv server-key.pem $DOCKER_CERT_PATH

# since the default user is vagrant and it can run docker without sudo
CLIENT_SIDE_CERT_PATH=/home/vagrant/.docker

mkdir -p $CLIENT_SIDE_CERT_PATH
cp $DOCKER_CERT_PATH/ca.pem $CLIENT_SIDE_CERT_PATH
mv cert.pem key.pem $CLIENT_SIDE_CERT_PATH

chown vagrant:vagrant $CLIENT_SIDE_CERT_PATH

chmod 0444 $CLIENT_SIDE_CERT_PATH/ca.pem
chmod 0444 $CLIENT_SIDE_CERT_PATH/cert.pem
chmod 0444 $CLIENT_SIDE_CERT_PATH/key.pem
chown vagrant:vagrant $CLIENT_SIDE_CERT_PATH/ca.pem
chown vagrant:vagrant $CLIENT_SIDE_CERT_PATH/cert.pem
chown vagrant:vagrant $CLIENT_SIDE_CERT_PATH/key.pem

chmod -v 0400 $DOCKER_CERT_PATH/ca.pem $DOCKER_CERT_PATH/server-cert.pem $DOCKER_CERT_PATH/server-key.pem

## Remove remaining files
cd
rm -rf $dir

# End of certs pair generation steps for TLS enabled docker daemon


build_tools/kickstarts/tls-certs-generation.sh
# Generate Certs for running TLS enabled docker daemon
#!/bin/bash
# title         : tls-certs-generation.sh
# description   : Certs pair generation script for configuring TLS enabled docker daemon
# author        : Brian Exelbierd <bexelbie@redhat.com>
# use           : sudo bash tls-certs-generation.sh
# version       : 0.1.1

# Generate Certificates to use with the docker daemon
# Instructions sourced from http://docs.docker.com/articles/https/

# Get the certificate location, i.e. setting the DOCKER_CERT_PATH variable
. /etc/sysconfig/docker

# randomString from http://utdream.org/post.cfm/bash-generate-a-random-string
# modified to echo value

function randomString {
        # if a param was passed, it's the length of the string we want
        if [[ -n $1 ]] && [[ "$1" -lt 20 ]]; then
                local myStrLength=$1;
        else
                # otherwise set to default
                local myStrLength=8;
        fi

        dd if=/dev/urandom bs=1 2>/dev/null | tr -dc '[:alnum:]' | dd bs=1 count=$myStrLength 2>/dev/null
}

# Get a temporary workspace
dir=`mktemp -d`
cd $dir

# Get a random password for the CA and save it
passfile=tmp.pass
password=$(randomString 10)
echo $password > $passfile

# Generate the CA
openssl genrsa -aes256 -passout file:$passfile -out ca-key.pem 2048
openssl req -new -x509 -passin file:$passfile -days 365 -key ca-key.pem -sha256 -out ca.pem -subj "/C=/ST=/L=/O=/OU=/CN=example.com"

# Generate Server Key and Sign it
openssl genrsa -out server-key.pem 2048
openssl req -subj "/CN=example.com" -new -key server-key.pem -out server.csr
# Allow from 127.0.0.1
extipfile=extfile.cnf
echo subjectAltName = IP:127.0.0.1 > $extipfile
openssl x509 -req -days 365 -in server.csr -CA ca.pem -CAkey ca-key.pem -CAcreateserial -out server-cert.pem -passin file:$passfile -extfile $extipfile

# Generate the Client Key and Sign it
openssl genrsa -out key.pem 2048
openssl req -subj '/CN=client' -new -key key.pem -out client.csr
extfile=tmp.ext
echo extendedKeyUsage = clientAuth > $extfile
openssl x509 -req -days 365 -in client.csr -CA ca.pem -CAkey ca-key.pem -CAcreateserial -out cert.pem -extfile $extfile -passin file:$passfile

# Clean up

# set the cert path as configured in /etc/sysconfig/docker

## Move files into place
mv ca.pem $DOCKER_CERT_PATH
mv server-cert.pem $DOCKER_CERT_PATH
mv server-key.pem $DOCKER_CERT_PATH

# since the default user is vagrant and it can run docker without sudo
CLIENT_SIDE_CERT_PATH=/home/vagrant/.docker

mkdir -p $CLIENT_SIDE_CERT_PATH
cp $DOCKER_CERT_PATH/ca.pem $CLIENT_SIDE_CERT_PATH
mv cert.pem key.pem $CLIENT_SIDE_CERT_PATH

chown vagrant:vagrant $CLIENT_SIDE_CERT_PATH

chmod 0444 $CLIENT_SIDE_CERT_PATH/ca.pem
chmod 0444 $CLIENT_SIDE_CERT_PATH/cert.pem
chmod 0444 $CLIENT_SIDE_CERT_PATH/key.pem
chown vagrant:vagrant $CLIENT_SIDE_CERT_PATH/ca.pem
chown vagrant:vagrant $CLIENT_SIDE_CERT_PATH/cert.pem
chown vagrant:vagrant $CLIENT_SIDE_CERT_PATH/key.pem

chmod -v 0400 $DOCKER_CERT_PATH/ca.pem $DOCKER_CERT_PATH/server-cert.pem $DOCKER_CERT_PATH/server-key.pem

## Remove remaining files
cd
echo rm -rf $dir


build_tools/build_scripts/do_vagrant_cbs_release.sh
#!/bin/sh
#To see all options in koji -p command check "koji -p cbs image-build --help"
COMMIT_ID=`git log | head -n 1 | cut -f 2 -d " "`

koji -p cbs image-build \
  centos-7-adb 1.5 atomic7-adb-common-el7 \
  http://mirror.centos.org/centos/7/os/x86_64/ x86_64 \
  --release=1 \
  --distro RHEL-7.0 \
  --ksver RHEL7 \
  --ksurl=https://github.com/projectatomic/adb-atomic-developer-bundle/blob/master/build_tools/kickstarts/centos-7-adb-vagrant.ks#$COMMIT_ID \
  --kickstart=./build_tools/kickstarts/centos-7-adb-vagrant.ks \
  --format=qcow2 \
  --format=vsphere-ova \
  --format=rhevm-ova \
  --ova-option vsphere_ova_format=vagrant-virtualbox \
  --ova-option rhevm_ova_format=vagrant-libvirt \
  --ova-option vagrant_sync_directory=/vagrant \
  --repo http://mirror.centos.org/centos/7/extras/x86_64/\
  --repo http://mirror.centos.org/centos/7/updates/x86_64/\
  --nowait \
  --disk-size=40

